From 0f8f369ee09eda116b5e971c07a99ab2f6fa51e3 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 27 Dec 2016 11:31:34 +0100
Subject: [PATCH] Enable SMPEN bit

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 arch/arm64/kernel/head.S | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/kernel/head.S b/arch/arm64/kernel/head.S
index 4b1abac3485a..ab0d56468721 100644
--- a/arch/arm64/kernel/head.S
+++ b/arch/arm64/kernel/head.S
@@ -487,6 +487,11 @@ ENTRY(kimage_vaddr)
  * booted in EL1 or EL2 respectively.
  */
 ENTRY(el2_setup)
+	/* Enable SMPEN bit */
+	mrs	x0, S3_1_c15_c2_1               // cpuactlr_el1
+	orr	x0, x0, #0x40			// set SMPEN bit
+	msr	S3_1_c15_c2_1, x0
+
 	mrs	x0, CurrentEL
 	cmp	x0, #CurrentEL_EL2
 	b.ne	1f
-- 
2.11.0

